### PR TITLE
Compatibility changes for python 3.11

### PR DESCRIPTION
**Pull request by jMyles**

### CODE REVIEW

**Comment by derekpierre** [comment]
Is this based over #3108 ?

**Comment by KPrasch** [comment]
This commit includes changes from the base branch 🤔

File: tests/unit/crypto/test_keccak_sanity.py [hunk 1]
```diff
@@ -37,15 +37,15 @@ def test_secure_random_range(self):
     def test_keccak_digest(self):
         data = b'this is a test'
 
-        digest1 = sha3.keccak_256(data).digest()
```
Comment (derekpierre) [inline tests/unit/crypto/test_keccak_sanity.py:40]: Does it make sense to store some well-known output from the `sha3` implementation as constants and then use for comparison, to confirm/reassure we are still backwards-compatible?

Comment (jMyles) [reply tests/unit/crypto/test_keccak_sanity.py]: ...I thought we decided we didn't care, and that we were OK with obseleting legacy fingerprints?

Comment (derekpierre) [reply tests/unit/crypto/test_keccak_sanity.py]: This test is only changing the library we use to compute the keccak_256 digest, no?  If so, we still compute the keccak digest so I was wondering if we want to ensure that the new library produces the same digest as the old library. Or is that excessive and/or irrelevant now?

Comment (KPrasch) [reply tests/unit/crypto/test_keccak_sanity.py]: Yep, I think it's okay to obsolete old fingerprints.

**Pull request closed without merging**